main.go

```diff
@@ -2,15 +2,45 @@ package main
 
 import (
 	"os"
+	"runtime/debug"
+	"strings"
 
 	"github.com/magnusbaeck/logstash-filter-verifier/v2/internal/app"
 )
 
-// GitSummary contains "git describe" output and is automatically
-// populated via linker options when building with govvv.
-var GitSummary = "(unknown)"
+func buildSetting(key string) (string, bool) {
+	info, _ := debug.ReadBuildInfo()
+	for _, kv := range info.Settings {
+		if kv.Key == key {
+			return kv.Value, true
+		}
+	}
+	return "", false
+}
 
 func main() {
-	exitCode := app.Execute(GitSummary, os.Stdout, os.Stderr)
+	var vcsinfo = []string{}
+	if vcs, found := buildSetting("vcs"); found {
+		vcsinfo = append(vcsinfo, vcs)
+	}
+	if revision, found := buildSetting("vcs.revision"); found {
+		if len(revision) > 7 {
+			revision = revision[0:7]
+		}
+		vcsinfo = append(vcsinfo, revision)
+	}
+	if time, found := buildSetting("vcs.time"); found {
+		vcsinfo = append(vcsinfo, time)
+	}
+	if modified, found := buildSetting("vcs.modified"); found {
+		if modified == "true" {
+			vcsinfo = append(vcsinfo, "dirty")
+		}
+	}
+	var version = "(unknown)"
+	if len(vcsinfo) > 0 {
+		version = strings.Join(vcsinfo, " ")
+	}
```

Comment:
One last suggestion, move this into its own function, such that it does not clutter `main`. Maybe something like

```golang
func version() string {
	var vcsinfo = []string{}
	if vcs, found := buildSetting("vcs"); found {
		vcsinfo = append(vcsinfo, vcs)
	}
	if revision, found := buildSetting("vcs.revision"); found {
		if len(revision) > 7 {
			revision = revision[0:7]
		}
		vcsinfo = append(vcsinfo, revision)
	}
	if time, found := buildSetting("vcs.time"); found {
		vcsinfo = append(vcsinfo, time)
	}
	if modified, found := buildSetting("vcs.modified"); found {
		if modified == "true" {
			vcsinfo = append(vcsinfo, "dirty")
		}
	}
	var version = "(unknown)"
	if len(vcsinfo) > 0 {
		version = strings.Join(vcsinfo, " ")
	}
	return version
}
```

Then this can be used like this:

```golang
exitCode := app.Execute(version(), os.Stdout, os.Stderr)
```

Reply:
Very good suggestion, thank you. I've adjusted my PR.